src/Providers/Models/Contracts/WithHttpTransporterInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Providers\Models\Contracts;
+
+use WordPress\AiClient\Providers\Http\Contracts\HttpTransporterInterface;
+
+/**
+ * Interface for models that require HTTP transport capabilities.
+ *
+ * Models implementing this interface can send HTTP requests
+ * to provider APIs using the configured HTTP transporter.
+ *
+ * @since n.e.x.t
+ */
+interface WithHttpTransporterInterface
```

Comment:
This is a duplicate of the same contract in `Providers\Http\Contracts`? I don't think that's needed - maybe here by accident? :)